### PR TITLE
Resource level filter logging feature added

### PR DESCRIPTION
**Pull request by shnrndk**
Fix https://github.com/wso2/api-manager/issues/1564
Test Case for this PR: https://github.com/wso2/product-apim/pull/13260

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/ExceptionCodes.java [hunk 1]
```diff
@@ -494,6 +494,7 @@ public enum ExceptionCodes implements ErrorHandler {
     LOGGING_API_NOT_FOUND(901400, "Requested Resource Not Found", 404, "Request API Not Found for context: %s"),
     LOGGING_API_INCORRECT_LOG_LEVEL(901401, "Bad Request", 400, "Log level should be either OFF, BASIC, STANDARD or FULL"),
     LOGGING_API_MISSING_DATA(901402, "Missing data", 400, "API context or log level is missing"),
+    LOGGING_API_RESOURCE_NOT_FOUND(901403, "Requested Resource Not Found", 400, "Requested API Resource Not Found"),
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/ExceptionCodes.java:497]: 404 for not found

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/LogUtils.java [hunk 2]
```diff
@@ -18,13 +18,19 @@
...
+import org.apache.synapse.api.ApiUtils;
+import org.apache.synapse.api.Resource;
+import org.apache.synapse.api.dispatch.DispatcherHelper;
+import org.apache.synapse.api.dispatch.RESTDispatcher;
 import org.apache.synapse.core.axis2.Axis2MessageContext;
 import org.wso2.carbon.apimgt.gateway.APIMgtGatewayConstants;
 import org.wso2.carbon.apimgt.impl.APIConstants;
 
-import java.util.Map;
+import java.util.*;
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/LogUtils.java:33]: Better not to use wild card imports. Importing specific class improves readability

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/LogUtils.java [hunk 3]
```diff
@@ -119,16 +125,99 @@ protected static String getTransportInURL(org.apache.synapse.MessageContext mess
...
+            if (!acceptableResources.isEmpty()) {
+                for (RESTDispatcher dispatcher : ApiUtils.getDispatchers()) {
+                    selectedResource = dispatcher.findResource(messageContext, acceptableResources);
+                    if (selectedResource != null) {
+                        DispatcherHelper helper = selectedResource.getDispatcherHelper();
+                        for (Map.Entry<Map<String, String>, String> entry : logProperties.entrySet()) {
+                            Map<String, String> key = entry.getKey();
+                            //if resource path is empty, proceeding with API level logs
+                            if (selectedApi.getContext().equals(key.get("context"))) {
+                                if (key.get("resourcePath") == null && key.get("resourceMethod") == null) {
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/LogUtils.java:166]: use constants previously defined

File: features/apimgt/org.wso2.carbon.apimgt.rest.api.service.catalog.feature/src/main/resources/runtimes/cxf3/snakeyaml-1.33.jar
Comment (chamilaadhi) [inline features/apimgt/org.wso2.carbon.apimgt.rest.api.service.catalog.feature/src/main/resources/runtimes/cxf3/snakeyaml-1.33.jar:1]: Why a dependency is added here?

**Comment by tharikaGitHub** [comment]
Closing as the changes are added from PR [1].

[1] https://github.com/wso2/carbon-apimgt/pull/12288

**Pull request closed without merging**